packages/salesforcedx-vscode-apex-replay-debugger/src/commands/quickLaunch.ts

```diff
@@ -79,6 +88,20 @@ export class QuickLaunch {
         testClass
       );
       const result: TestResult = await testService.runTestSynchronous(payload);
+
+      // create apex test result files
+      if (workspace && workspace.workspaceFolders) {
+        const apexTestResultsPath = getTestResultsFolder(
+          getRootWorkspacePath(),
+          'apex'
+        );
+        await testService.writeResultFiles(
+          result,
+          { dirPath: apexTestResultsPath, resultFormats: [ResultFormat.json] },
+          retrieveTestCodeCoverage()
+        );
+      }
```

Comment:
It looks like the test sidebar isn't getting updated after the first run due to how the result files are generated for synchronous test runs. Synchronous test runs don't have a test run id - so after the first test result file is generated, `test-result-.json`, each subsequent run just _updates_ the content of that file. We could change the fileWatcher to fire on the `onDidChange` event to get around this, otherwise does this command need to use a synchronous test run? We could also switch to asynchronous test runs to avoid this.

On a sidenote, I'll file a bug on the apex library because it looks like we're appending an extra `-` onto the end of the result file

Reply:
Yeah, I'll have to think about that. I'm leaning towards changing the watchers.

Specific to the additional `-` character, are you talking about the content or the name of the file ? The name of the file should be `test-result-<test result id>.json`, since sync test runs don't generate test result ids the current behavior is correct.

Reply:
Sounds good, I think changing the watchers works!

I was referring to the name of the file, right now it gets created with the name `test-result-.json`. This makes sense, but it looks like the CLI currently creates it as just `test-result.json`. We can add this to the list of changes coming in the future, but I think we'll need to switch to the CLI version for now since we aren't touching the result files with this first release.

Reply:
Sounds good 👍